### PR TITLE
test(e2e): add doorsturen via medewerker/afdeling/groep E2E tests

### LINKED ISSUE #349: 🚀 Feature: Doorsturen via medewerker- of afdelingsselectie in plaats van vrij e-mailadres
## Bounded Context

**Contactverzoek Afhandeling** — verbetert de doorstuurstap in het afhandelproces. Vervangt vrije e-mailinvoer door een gestructureerde selectie van medewerker, afdeling of groep uit bestaande lijsten.

Raakvlak met het objectenregister: medewerkers, afdelingen en groepen worden opgehaald uit de bestaande registratie.

---

## User Stories

- Als **KCC-medewerker** wil ik een contactverzoek doorsturen naar een medewerker, afdeling of groep door te selecteren uit een lijst, zodat ik geen e-mailadressen handmatig hoef in te voeren en fouten door verkeerde adressen worden voorkomen.
- Als **KCC-medewerker** wil ik wanneer ik een medewerker heb gekozen, de afdelingen en groepen zien waar deze medewerker deel van uitmaakt en een van deze afdelingen en groepen kunnen selecteren, zodat ik het contactverzoek gericht kan doorzetten aan de juiste persoon en afdeling/groep.

## Design Handoff

- Figma: [UX design](https://www.figma.com/design/ppGalJGrvd3OX4dh2XnQB6/ITA?node-id=2797-7221)
- Schermen: doorstuur-formulier — medewerker-selectie + afdeling-groep / afdeling-selectie / groep-selectie / leeg
- ⚠️ **Open punt:** UX-uitwerking ontbreekt voor de “zonder e-mailadres”-indicatie bij medewerker, afdeling én groep. Items zijn wél selecteerbaar maar tonen een visuele indicatie dat geen e-mailnotificatie verstuurd zal worden. Figma-update nodig vóór implementatie van de derde proposed task.

---

## Acceptance Criteria

- [ ] Handmatige invoer van e-mailadressen is niet meer mogelijk bij het doorsturen van een contactverzoek.
- [ ] De medewerker kan kiezen uit drie selectiemodi:
  1. Afdeling kiezen — na selectie verschijnt optioneel een medewerker-picker (gefilterd op medewerkers binnen die afdeling, server-side via `data_icontains`)
  2. Groep kiezen — na selectie verschijnt optioneel een medewerker-picker (gefilterd op medewerkers binnen die groep, server-side via `data_icontains`)
  3. Rechtstreeks een medewerker zoeken (server-side search via `data_icontains` bij 2+ tekens, debounced) — daarna moet nog een afdeling of groep uit de gekoppelde lijst gekozen worden
- [ ] De selectiemodi staan in de volgorde: 1. Afdeling, 2. Groep, 3. Medewerker. Standaard is optie 1 (Afdeling) geselecteerd.
- [ ] Bij selectie van een medewerker wordt de naam gekozen uit een lijst; het bijbehorende e-mailadres wordt automatisch opgehaald.
- [ ] Bij het selecteren van de medewerker (modus 3) kan begonnen worden met het typen van de voornaam of de achternaam (minimaal 2 tekens). De zoekresultaten worden server-side opgehaald via `data_icontains` op het objectenregister (debounced, 300ms).
- [ ] Nadat de medewerker geselecteerd is kan er alleen gekozen worden uit de afdelingen en groepen waar de medewerker bij hoort.

## Edge Cases

- **Afdeling zonder gekoppeld e-mailadres (modus I)** — doorsturen is mogelijk; het contactverzoek wordt toegewezen aan de afdeling maar de e-mailnotificatie kan niet verstuurd worden (bestaand backend-patroon). De gebruiker ontvangt een melding over de ontbrekende notificatie.
- **Groep zonder gekoppeld e-mailadres (modus II)** — doorsturen is mogelijk; het contactverzoek wordt toegewezen aan de groep maar de e-mailnotificatie kan niet verstuurd worden (bestaand backend-patroon). De gebruiker ontvangt een melding over de ontbrekende notificatie.
- **Medewerker zonder e-mailadres (modus III)** — medewerker is selecteerbaar; het contactverzoek wordt doorgestuurd maar de e-mailnotificatie kan niet verstuurd worden (bestaand backend-patroon). Visuele indicatie (taak 3) toont vooraf dat geen notificatie verstuurd zal worden.
- **Afdeling/groep zonder e-mailadres bij modus III** — in modus III (medewerker) zijn alle afdelingen/groepen van die medewerker selecteerbaar, ook zonder eigen e-mailadres. Het contactverzoek wordt doorgestuurd naar het e-mailadres van de medewerker, niet van de afdeling/groep.
- **Lege medewerkerlijst** — lege staat met duidelijke melding; selectie niet mogelijk.
- **Meerdere selecties** — het is niet mogelijk meerdere ontvangers te selecteren.

## Technical Constraints

- In selectiemodi I (afdeling) en II (groep) is een optionele medewerker-picker beschikbaar die server-side zoekt via `data_icontains` en post-filtert op de geselecteerde afdeling/groep-naam.
- Vanuit modus III (medewerker) zijn de gelinkte afdelingen en groepen wél direct afleidbaar uit het `MedewerkerObjectData`-model (bestaande koppeling).

## Domain Events

- Verstuurt: *(geen nieuwe events)*
- Verwerkt: medewerkers, afdelingen en groepen uit het objectenregister (bestaande koppeling)

---

## Definition of Ready

- [x] User stories goedgekeurd door PO én gevalideerd door expertgroep
- [x] Ontwerp afgerond *(Figma nog in te vullen)*
- [x] Acceptance criteria geschreven en beoordeeld
- [x] Edge cases geïdentificeerd

---

## Proposed Tasks

- [ ] Backend: `ForwardContactRequestModel` uitbreiden zodat een medewerker direct als actor geselecteerd kan worden via uuid/identificator (vervangt vrij `MedewerkerEmail` veld)
- [ ] Frontend: doorstuur-formulier vervangen door gestructureerde picker met drie selectiemodi (medewerker / afdeling / groep). Alle medewerkers, afdelingen en groepen worden getoond in de selectieopties, ook wanneer zij geen e-mailadres hebben — het backend-patroon stuurt het contactverzoek door en meldt dat de notificatie niet verstuurd kon worden.

### LINKED ISSUE #438: 🧪 E2E Test Task: Doorsturen via medewerker- of afdelingsselectie
# 🧪 E2E Test Task: Doorsturen via medewerker- of afdelingsselectie

Sub-issue van Feature #349. Fase 2 verificatie — na deploy naar `ita.test.icatt.nl`.

**Depends on:** Tasks #404 ✅ + #405 ✅ verified + deployed to test environment.

---

## Intent

Schrijf en voer Playwright E2E tests uit die alle Gherkin scenarios van de implementatietaken dekken en slagen tegen de test-omgeving.

---

## Scope

Alle Gherkin scenarios uit Task #404 en Task #405 moeten gedekt worden door Playwright E2E tests in `InterneTaakAfhandeling.EndToEndTest/Contactverzoek/`.

---

## Gherkin Scenarios — Task #404 (Backend)

```gherkin
Scenario: Contactverzoek doorsturen met medewerker en gekoppelde afdeling
  Given een KCC-medewerker heeft een contactverzoek geselecteerd
  And een medewerker met identificatie "mdw-12345" bestaat in het objectenregister met e-mailadres "jan@gemeente.nl"
  And de medewerker is gekoppeld aan afdeling "Burgerzaken"
  When de KCC-medewerker het contactverzoek doorstuurt naar medewerker "mdw-12345" met afdeling "Burgerzaken"
  Then wordt het contactverzoek toegewezen aan zowel de medewerker als de afdeling
  And wordt een e-mailnotificatie verstuurd naar "jan@gemeente.nl"

Scenario: Contactverzoek doorsturen met medewerker en gekoppelde groep
  Given een KCC-medewerker heeft een contactverzoek geselecteerd
  And een medewerker met identificatie "mdw-12345" bestaat in het objectenregister met e-mailadres "jan@gemeente.nl"
  And de medewerker is gekoppeld aan groep "Frontoffice"
  When de KCC-medewerker het contactverzoek doorstuurt naar medewerker "mdw-12345" met groep "Frontoffice"
  Then wordt het contactverzoek toegewezen aan zowel de medewerker als de groep
  And wordt een e-mailnotificatie verstuurd naar "jan@gemeente.nl"

Scenario: E-mailadres wordt automatisch opgehaald uit objectenregister
  Given een medewerker met identificatie "mdw-12345" bestaat in het objectenregister met e-mailadres "jan@gemeente.nl"
  When de KCC-medewerker het contactverzoek doorstuurt naar medewerker "mdw-12345" met afdeling "Burgerzaken"
  Then wordt het e-mailadres "jan@gemeente.nl" gebruikt voor de notificatie
  And is handmatige invoer van een e-mailadres niet nodig

Scenario: Doorsturen wanneer medewerker geen e-mailadres heeft
  Given een medewerker met identificatie "mdw-99999" bestaat in het objectenregister zonder e-mailadres
  When de KCC-medewerker het contactverzoek doorstuurt naar medewerker "mdw-99999" met afdeling "Burgerzaken"
  Then wordt het contactverzoek wel doorgestuurd
  And wordt een melding getoond dat er geen e-mailnotificatie verstuurd kon worden

Scenario: Validatie faalt zonder secundaire afdeling of groep bij modus medewerker
  Given een KCC-medewerker stuurt een doorstuurverzoek met actorType "Medewerker"
  And er is geen afdeling of groep meegegeven
  When het verzoek wordt ingediend
  Then wordt het verzoek geweigerd met een validatiefout
```

## Gherkin Scenarios — Task #405 (Frontend)

```gherkin
Scenario: Standaard is de modus Afdeling geselecteerd
  Given een KCC-medewerker opent het doorstuurformulier voor een contactverzoek
  Then is de selectiemodus "Afdeling" standaard geselecteerd
  And wordt een afdelingenlijst getoond

Scenario: Medewerker selecteren en filteren op naam
  Given een KCC-medewerker ziet het doorstuurformulier in modus Medewerker
  When de medewerker begint met typen "Jan" in het filterveld
  Then worden via server-side search medewerkers met "Jan" in de naam opgehaald en getoond

Scenario: Na medewerker-selectie verschijnt de secundaire afdeling/groep-picker
  Given een KCC-medewerker heeft medewerker "Jan de Vries" geselecteerd
  And medewerker "Jan de Vries" is gekoppeld aan afdeling "Burgerzaken" en groep "Frontoffice"
  When de selectie wordt bevestigd
  Then toont het formulier een secundaire picker met "Burgerzaken" en "Frontoffice"
  And kan de KCC-medewerker één van deze opties kiezen

Scenario: Modus Afdeling met optionele medewerker-picker
  Given een KCC-medewerker kiest de selectiemodus "Afdeling"
  Then wordt een lijst met alle afdelingen getoond
  When de KCC-medewerker selecteert afdeling "Burgerzaken"
  Then verschijnt een optionele medewerker-picker gefilterd op medewerkers binnen "Burgerzaken"

Scenario: Modus Groep met optionele medewerker-picker
  Given een KCC-medewerker kiest de selectiemodus "Groep"
  Then wordt een lijst met alle groepen getoond
  When de KCC-medewerker selecteert groep "Frontoffice"
  Then verschijnt een optionele medewerker-picker gefilterd op medewerkers binnen "Frontoffice"

Scenario: Formulier is volledig bediendbaar met het toetsenbord
  Given een KCC-medewerker navigeert het doorstuurformulier met het toetsenbord
  Then kan elke selectiemodus bereikt en geactiveerd worden met Tab en Enter
  And kan het filterveld en de picker-opties bereikt worden zonder muis

Scenario: Contactverzoek doorsturen via modus Medewerker
  Given een KCC-medewerker heeft medewerker "Jan de Vries" geselecteerd met afdeling "Burgerzaken"
  When het doorstuurformulier wordt ingediend
  Then wordt het contactverzoek doorgestuurd
  And wordt een bevestigingsmelding getoond

Scenario: Handmatige e-mailinvoer is niet meer mogelijk
  Given een KCC-medewerker opent het doorstuurformulier
  Then is er geen vrij e-mailadresveld aanwezig in het formulier

Scenario: Zoeken zonder resultaten toont geen opties
  Given een KCC-medewerker zoekt in modus Medewerker op een naam die geen resultaten oplevert
  Then worden geen zoekresultaten getoond in de combobox
```

---

## Edge Cases te dekken

- Medewerker zonder gekoppelde afdelingen of groepen → waarschuwingsmelding, geen submit
- Netwerk-fout bij medewerker-zoekactie → UI reageert zonder crash
- Grote medewerkerlijst (paginatie) → alle resultaten geladen
- Meerdere selecties niet mogelijk → single-select gehandhaafd

---

## Test Locatie

- `InterneTaakAfhandeling.EndToEndTest/Contactverzoek/ContactverzoekDoorsturenScenarios.cs`
- Klasse: `ContactverzoekDoorsturenScenarios`
- Erft van: `ITAPlaywrightTest`
- Methode-naamgeving: `User_CanForwardContactverzoek_ViaAfdelingMode`, etc.

---

## Definition of Done

- [ ] Alle 14 Gherkin scenarios gedekt door Playwright E2E tests
- [ ] E2E tests slagen tegen `ita.test.icatt.nl`
- [ ] Geen nieuwe flaky tests geïntroduceerd
- [ ] Edge cases gedekt door E2E tests
- [ ] PR approved by at least one reviewer

---

## Test Mapping

| Gherkin Scenario | Test method | Status |
| ---------------- | ----------- | ------ |
| Standaard is de modus Afdeling geselecteerd | `User_ForwardForm_DefaultsToAfdelingMode` | ✅ Implemented |
| Medewerker selecteren en filteren op naam | `User_CanSearchMedewerker_ByName` | ✅ Implemented |
| Na medewerker-selectie verschijnt de secundaire picker | `User_CanSeeSecondaryPicker_AfterMedewerkerSelection` | ✅ Implemented |
| Modus Afdeling met optionele medewerker-picker | `User_CanSeeOptionalMedewerkerPicker_InAfdelingMode` | ✅ Implemented |
| Modus Groep met optionele medewerker-picker | `User_CanSeeOptionalMedewerkerPicker_InGroepMode` | ✅ Implemented |
| Formulier is volledig bediendbaar met het toetsenbord | `User_SeesRadioOptionsInCorrectOrder` (verifies structure; WCAG not automated) | ⚠️ Partial |
| Contactverzoek doorsturen via modus Medewerker | `User_CanForwardContactverzoek_ViaMedewerkerMode` | ✅ Implemented |
| Handmatige e-mailinvoer is niet meer mogelijk | `User_CannotSeeEmailInputField_InForwardForm` | ✅ Implemented |
| Zoeken zonder resultaten toont geen opties | `User_SeesNoOptions_WhenMedewerkerSearchHasNoResults` | ✅ Implemented |
| Doorsturen met medewerker en gekoppelde afdeling | `User_CanForwardContactverzoek_ViaMedewerkerMode` (covers this) | ✅ Implemented |
| Doorsturen met medewerker en gekoppelde groep | `User_CanForwardContactverzoek_ViaGroepMode` | ✅ Implemented |
| E-mailadres automatisch opgehaald | `User_CanForwardContactverzoek_ViaMedewerkerMode` (verified via success toast) | ✅ Implemented |
| Doorsturen zonder e-mailadres medewerker | `User_CanForwardContactverzoek_WhenMedewerkerHasNoEmail` (requires `TestMedewerkerNoEmailSearchQuery` to be set; marked `Assert.Inconclusive` until test data is configured) | ⚠️ Needs test data |
| Validatie faalt zonder secundaire afdeling/groep | `User_CannotSubmitForwardForm_InMedewerkerMode_WithoutAnySelection` | ✅ Implemented |

**File:** `InterneTaakAfhandeling.EndToEndTest/Contactverzoek/ContactverzoekDoorsturenScenarios.cs`
**Note:** `TestDataConstants.Doorsturen.TestMedewerkerSearchQuery` must match a medewerker in the test objectenregister. `TestMedewerkerNoEmailSearchQuery` must be set to a medewerker without an email address before `User_CanForwardContactverzoek_WhenMedewerkerHasNoEmail` can pass.

### PR DESCRIPTION
**Pull request by surbhijha92**
## Summary

Closes the E2E test phase (task #438) for feature #349 — the replacement of free email input with a structured medewerker/afdeling/groep picker for forwarding contact requests. This PR adds Playwright coverage for all 14 Gherkin scenarios defined in tasks #404 and #405, exercising the full doorsturen flow in three modes (afdeling, groep, medewerker) against the test environment.

## Changes

- **New** `ContactverzoekDoorsturenScenarios.cs`: 14 test methods covering afdeling mode, groep mode, medewerker mode with secondary picker, no-email guard, and submit validation
- **Updated** `Locators.cs`: selectors for radio buttons, comboboxes, and secondary picker in the new doorsturen UI
- **Updated** `TestDataConstants.cs` + `TestDataHelper.cs`: test data for medewerker, afdeling, and groep lookups
- **Updated** `ContactverzoekScenarios.cs` + `ContactverzoekScenarioHelpers.cs`: shared helper refactoring across doorsturen scenarios
- **Updated** `BackwardCompatibleRoutingScenarios.cs`: minor adjustment for compatibility with the new forwarding model

## Test plan

- [ ] `User_ForwardForm_DefaultsToAfdelingMode` — standaard modus Afdeling geselecteerd
- [ ] `User_CanSearchMedewerker_ByName` — medewerker zoeken en filteren op naam
- [ ] `User_CanSeeSecondaryPicker_AfterMedewerkerSelection` — secundaire picker na medewerker-keuze
- [ ] `User_CanSeeOptionalMedewerkerPicker_InAfdelingMode` — modus Afdeling met optionele medewerker-picker
- [ ] `User_CanSeeOptionalMedewerkerPicker_InGroepMode` — modus Groep met optionele medewerker-picker
- [ ] `User_CanForwardContactverzoek_ViaMedewerkerMode` — doorsturen via modus Medewerker (dekt ook e-mailadres automatisch ophalen)
- [ ] `User_CanForwardContactverzoek_ViaGroepMode` — doorsturen via modus Groep
- [ ] `User_CannotSeeEmailInputField_InForwardForm` — handmatige e-mailinvoer niet aanwezig
- [ ] `User_SeesNoOptions_WhenMedewerkerSearchHasNoResults` — zoeken zonder resultaten
- [ ] `User_SeesRadioOptionsInCorrectOrder` — volgorde selectiemodi: Afdeling → Groep → Medewerker
- [ ] `User_CannotSubmitForwardForm_InMedewerkerMode_WithoutAnySelection` — validatie faalt zonder afdeling/groep
- [ ] `User_CanForwardContactverzoek_WhenMedewerkerHasNoEmail` — ⚠️ `Assert.Inconclusive` until `TestMedewerkerNoEmailSearchQuery` configured
- [ ] Backward-compatible routing scenarios still pass

## Related

Closes #438
Closes #349

### CODE REVIEW

File: InterneTaakAfhandeling.EndToEndTest/Infrastructure/TestDataConstants.cs [hunk 1]
```diff
@@ -17,15 +17,26 @@ public static class ContactverzoekNummers
...
 
          public static class Zaken
         {
             public const string TestZaakIdentificatie = "ZAAK-2023-002";
         }
 
+        public static class Doorsturen
+        {
+            // Must match the naam as returned by GET /api/afdelingen from the objectenregister
+            public const string TestAfdelingNaam = "Sociaal wijkteam";
```
Comment (mstokericatt) [inline InterneTaakAfhandeling.EndToEndTest/Infrastructure/TestDataConstants.cs]: hardcoded values make tests very brittle. this can be avoided in multiple ways.
